test/jest/prometheus.spec.ts

```diff
@@ -26,18 +26,34 @@ describe("Prometheus and Alertmanager", () => {
   });
 
   test("alert manager should be firing watchdog alert", async () => {
-    // fetch active alerts with alertname="Watchdog"
-    const response = await fetch(
-      `${alertmanagerProxy.url}/api/v2/alerts/groups?filter=alertname%3D%22Watchdog%22&silenced=false&inhibited=false&active=true`,
-    );
-    expect(response.status).toBe(200);
+    // The Watchdog configuration uses group_interval: 5m and group_wait: 30s. This test might be executed too quickly
+    // to catch this. Therefore, we retry this test over the course of 6 mins (36 retries of 10s each). The absolute
+    // test timeout is set to 7 minutes to accommodate extra execution delays.
+    const maxRetries = 36;
+    const delay = 10000;
+    let success = false;
```

Comment:
I'm in general okay with this, although I don't love potentially adding 7 mins to our CI. Given the config you mentioned, should this really only retry for ~1 min? If the initial wait is only 30s, I'm wondering if we'd have confidence that there should be a first alert sent through within that first group? But I may be misunderstanding these parameters and how this works.

I also wonder if there's a more direct way/different way to check prom/alertmanager connectivity alternatively here? Just wary of potentially adding extra CI time, although I recognize that these retries should prevent "false failures", actually saving time in the end ideally.

Reply:
Changed the approach as we discussed during the Slack chat.